projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/experiment/Experiment.java

```diff
@@ -159,6 +159,15 @@ public void addSpecimens(Collection<Specimen> specimens) {
     this.specimens.addAll(missingSpecimens);
   }
 
+  public void removeAllExperimentalVariables() {
+    removeAllExperimentalGroups();
+    experimentalDesign.removeAllExperimentalVariables();
+  }
+
+  public void removeAllExperimentalGroups() {
```

Comment:
Sorry but this method produces a ConcurrentModificationException
```2
023-07-18 13:11:27,821 ERROR [http-nio-8080-exec-10] l.q.d.e.UiExceptionHandler: null
java.util.ConcurrentModificationException: null
```
The consumer edits the list itself. You could circumvent this by first collecting the ids and iterating over the ids instead.